File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
        left := 0
        currSum := 0
        minLen := math.MaxInt
        for right := 0; right < len(nums); right++ {
            currSum += nums[right]
            for currSum >= target {
                minLen = min(minLen, right-left+1)
                currSum -= nums[left]
                left++
            }
        }
        if minLen == math.MaxInt {
            return 0
        }
        return minLen
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        minLen = float('inf')
        left = 0
        currSum = 0

        for right in range(len(nums)):
            currSum += nums[right]
            while currSum >= target:
                minLen = min(minLen, right - left + 1)
                currSum -= nums[left]
                left += 1
        if minLen == float('inf'):
            return 0
        return minLen